sanitize method for Block Tool seems to be typed wrong

Case 1:

In [Block Tool types](https://github.com/codex-team/editor.js/blob/next/types/tools/block-tool.d.ts), sanitize should return SanitizerConfig (which is object with tag names as keys). [Examples in SanitizerConfig](https://github.com/codex-team/editor.js/blob/next/types/configs/sanitizer-config.d.ts) are also written for fields with tag names as keys.

Case 2:

In [documentation](https://editorjs.io/tools-api/#sanitize), it returns object with 'url' and 'caption' keys.
Same behavior in other official plugins, such as [paragraph](https://github.com/editor-js/paragraph/blob/master/src/index.ts), [quote](https://github.com/editor-js/quote/blob/master/src/index.ts) etc.
And unfortunately, in that situation I can't use function as value for tag name, because tag name here becomes typed as 'attr'.

Editor.js version: 2.30.8

I can't understand which of that cases is right or wrong.
Is there any correct way to use function for sanitizing tag?

@neSpecc 

I'm trying to do that

<img width="1487" height="605" alt="Image" src="https://github.com/user-attachments/assets/db7ad39c-262f-40ab-9b67-4dac5cfc5f0c" />

and receiving TypeScript error TS2322: Type '(el: any) => boolean' is not assignable to type 'string | boolean'

--- a/docs/CHANGELOG.md
+++ b/docs/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Changelog
 
+### 2.31.6
+
+- `Fix` - Widen `sanitize` type on `BlockTool` and `BaseToolConstructable` to accept per-field `SanitizerConfig`
+
 ### 2.31.5
 
 - `Fix` - Handle __Ctrl + click__ on links with inline styles applied (e.g., bold, italic)

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@editorjs/editorjs",
-  "version": "2.31.5",
+  "version": "2.31.6",
   "description": "Editor.js — open source block-style WYSIWYG editor with JSON output",
   "main": "dist/editorjs.umd.js",
   "module": "dist/editorjs.mjs",

--- a/types/tools/block-tool.d.ts
+++ b/types/tools/block-tool.d.ts
@@ -13,9 +13,15 @@ import { MenuConfig } from './menu-config';
  */
 export interface BlockTool extends BaseTool {
   /**
-   * Sanitizer rules description
+   * Sanitizer rules description.
+   *
+   * @example Flat config (tag-level rules applied to all output)
+   * { b: true, a: { href: true } }
+   *
+   * @example Per-field config
+   * { text: { br: true, b: true }, caption: { b: true, i: true } }
    */
-  sanitize?: SanitizerConfig;
+  sanitize?: SanitizerConfig | Record<string, SanitizerConfig>;
 
   /**
    * Process Tool's element in DOM and return raw data

--- a/types/tools/tool.d.ts
+++ b/types/tools/tool.d.ts
@@ -37,9 +37,12 @@ export interface BaseToolConstructable {
   isInline?: boolean;
 
   /**
-   * Tool`s sanitizer configuration
+   * Tool`s sanitizer configuration.
+   *
+   * For Block Tools, can be a Record mapping data field names to their SanitizerConfig.
+   * For Inline Tools, should be a flat SanitizerConfig with tag names as keys.
    */
-  sanitize?: SanitizerConfig;
+  sanitize?: SanitizerConfig | Record<string, SanitizerConfig>;
 
   /**
    * Title of Inline Tool.

